### PR TITLE
refactor: TrackingVolume and Volume interface + functionality

### PR DESCRIPTION
**Pull request by paulgessinger**
This PR:
- Marks all of the Gen1 interface of `TrackingVolume` as such, to indicate what we'll supersede in the future
- Makes a slight adjustment to [`TransformRange`](https://github.com/acts-project/acts/pull/3060), for
- Add internal vector of uniquely owned child `TrackingVolume`s in `TrackingVolume`, accessible in a const-preserving way through `TransformRange`.
- Updates the `TrackingVolume` constructors, eliminating one convenience one we don't need, and start adding ones we'll use for Gen3

Blocked by:
- #3060

### CODE REVIEW

File: Core/include/Acts/Geometry/TrackingVolume.hpp [hunk 1]
```diff
@@ -406,6 +450,25 @@ class TrackingVolume : public Volume {
...
+  /// Array of Volumes inside the Volume when actin as container
+  std::shared_ptr<const TrackingVolumeArray> m_confinedVolumes = nullptr;
+
+  /// confined dense
+  MutableTrackingVolumeVector m_confinedDenseVolumes;
+
+  /// Volumes to glue Volumes from the outside
+  GlueVolumesDescriptor* m_glueVolumeDescriptor{nullptr};
+
+  /// @}
```
Comment (andiwand) [inline Core/include/Acts/Geometry/TrackingVolume.hpp:469]: where does that block start?

File: Core/include/Acts/Geometry/Volume.hpp [hunk 2]
```diff
@@ -29,7 +29,7 @@ class VolumeBounds;
 /// the interface for inherited Volume classes
 /// regarding the geometrical information.
 
-class Volume : public virtual GeometryObject {
+class Volume : public GeometryObject {
```
Comment (andiwand) [inline Core/include/Acts/Geometry/Volume.hpp:32]: 💎 🗡️

File: Core/include/Acts/Geometry/Volume.hpp [hunk 3]
```diff
@@ -98,6 +103,8 @@ class Volume : public virtual GeometryObject {
   Vector3 binningPosition(const GeometryContext& gctx,
                           BinningValue bValue) const override;
 
+  bool operator==(const Volume& other) const;
```
Comment (andiwand) [inline Core/include/Acts/Geometry/Volume.hpp:106]: should we also define `!=`?

**Review by andiwand — DISMISSED** [review]

File: Core/include/Acts/Geometry/TrackingVolume.hpp (hunk 1 above)
Comment (paulgessinger) [reply Core/include/Acts/Geometry/TrackingVolume.hpp:469]: Line 314

File: Core/include/Acts/Geometry/Volume.hpp (hunk 2 above)
Comment (paulgessinger) [reply Core/include/Acts/Geometry/Volume.hpp:32]: Not sure when this was ever diamond...

File: Core/include/Acts/Geometry/Volume.hpp (hunk 3 above)
Comment (paulgessinger) [reply Core/include/Acts/Geometry/Volume.hpp:106]: Sure.

File: Core/include/Acts/Geometry/TrackingVolume.hpp [hunk 4]
```diff
@@ -113,20 +120,6 @@ class TrackingVolume : public Volume {
...
-  /// @param transform is the global 3D transform to position the volume in
-  /// space
-  /// @param volbounds is the description of the volume boundaries
-  /// @param containedVolumeArray are the static volumes that fill this volume
-  /// @param volumeName is a string identifier
-  TrackingVolume(const Transform3& transform,
-                 std::shared_ptr<VolumeBounds> volbounds,
-                 const std::shared_ptr<const TrackingVolumeArray>&
-                     containedVolumeArray = nullptr,
-                 const std::string& volumeName = "undefined");
```
Comment (andiwand) [inline Core/include/Acts/Geometry/TrackingVolume.hpp:128]: is this a breaking change?

**Review by andiwand — DISMISSED** [review]
looks good to me 👍 

should this be marked as breaking?

File: Core/include/Acts/Geometry/TrackingVolume.hpp (hunk 4 above)
Comment (paulgessinger) [reply Core/include/Acts/Geometry/TrackingVolume.hpp]: It's back now, so not breaking anymore imo @andiwand.

**Review by andiwand — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [12b861157bab5bbecccdac3c6a847c4c6eb75bcd]
### :white_check_mark: All tests successful

|status|job|report|
|------|---|------|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305677)||
|:green_circle: | [test_ActsEFTrackFit](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305676)||
|:green_circle: | [test_ActsPersistifySeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305675)||
|:green_circle: | [test_ActsBenchmarkWithSpot](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305674)||
|:green_circle: | [test_ActsAnalogueClustering](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305673)||
|:green_circle: | [test_ActsConversionWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305672)||
|:green_circle: | [test_ActsWorkflowFastTracking](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305671)||
|:green_circle: | [test_ActsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305670)||
|:green_circle: | [test_ActsValidateAmbiguityResolution](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305669)||
|:green_circle: | [test_ActsValidateResolvedTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305668)||
|:green_circle: | [test_ActsValidateTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305667)||
|:green_circle: | [test_ActsValidateActsCoreSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305666)||
|:green_circle: | [test_ActsValidateActsSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305665)||
|:green_circle: | [test_ActsValidateSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305664)||
|:green_circle: | [test_ActsValidateOrthogonalSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305663)||
|:green_circle: | [test_ActsValidateClusters](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305662)||
|:green_circle: | [test_ActsPersistifyEDM](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305661)||
|:green_circle: | [test_ActsGSFRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305660)||
|:green_circle: | [test_ActsKfRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305659)||
|:green_circle: | [test_ActsExtrapolationAlgTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305658)||
|:green_circle: | [test_ActsITkTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305657)||
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305656)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305655)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305654)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305653)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305652)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305651)||
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38305650)||

**Pull request merged**